### PR TITLE
[lit-next] Rename hydrate & hydrate-support to "experimental"

### PR DESCRIPTION
**Pull request by kevinpschaaf**
We could also consider adding a top-level `console.warn` in `hydrate.js` (since `hydrate-support` also imports it) to the nature of:
> Lit hydration support is experimental. Experimental modules may undergo breaking changes within otherwise non-major releases.

WDYT?

### CODE REVIEW

**Comment by abdonrd** [comment]
Isn't everything in the `labs` folder considered experimental?

**Comment by kevinpschaaf** [comment]
> Isn't everything in the labs folder considered experimental?

Yes, however the `lit/hydrate.js` and `lit/hydrate-support.js` modules aren't themselves in labs, and decoupling them enough to move to a separate package would cause us to make more core library symbols public (and non-minifiable) than we'd like.

**Comment by abdonrd** [comment]
> > Isn't everything in the labs folder considered experimental?
> 
> Yes, however the `lit/hydrate.js` and `lit/hydrate-support.js` modules aren't themselves in labs, and decoupling them enough to move to a separate package would cause us to make more core library symbols public (and non-minifiable) than we'd like.

Oh, you're totally right! I don't know why I thought everything was in labs. Thanks!

**Comment by kevinpschaaf** [comment]
Yeah, the server-side SSR module itself is in labs, so you can't use hydrate without using `@lit-labs/ssr`, but as a pragmatic decision we're leaving hydration code in the core, and just want to be clear that we don't consider those API's to be totally stable yet, although the lit package itself will otherwise be.

File: packages/labs/ssr-client/package.json [hunk 1]
```diff
@@ -33,7 +33,7 @@
     "build:watch": "rollup -c --watch",
     "build:ts": "tsc --build && treemirror development . '**/*.d.ts{,.map}'",
     "build:ts:watch": "tsc --build --watch",
-    "clean": "rm -rf {index,hydrate-support,hydrate}.{js,js.map,d.ts} directives/ development/ test/ *.tsbuildinfo",
+    "clean": "rm -rf {index}.{js,js.map,d.ts} directives/ development/ test/ *.tsbuildinfo",
```
Comment (sorvell) [inline packages/labs/ssr-client/package.json:36]: Was hydrate there mistakenly before?

**Review by sorvell — APPROVED** [review]
Looks good. Just one question.

File: packages/labs/ssr-client/package.json (hunk 1 above)
Comment (kevinpschaaf) [reply packages/labs/ssr-client/package.json:36]: Yeah, vestigial from when I tried moving them there earlier, missed removing it.

**Review by justinfagnani — APPROVED** [review]

**Pull request merged**